### ISSUE TITLE
`msbuild /restore` with relative path to project file fails

### ISSUE THREAD

**Issue by dsplaisted** [Microsoft/msbuild#3926]
I ran into this using the locally built bootstrapped version of MSBuild from the master branch.  So it's possible it's a bug in the bootstrapped version of MSBuild, but not in the real product.  Here's what happens:

Run `msbuild restore <relative\path\to\project.csproj`.  In my example:

    C:\git\msbuild>artifacts\Debug\bootstrap\net472\MSBuild\Current\Bin\MSBuild.exe /restore ..\..\Users\daplaist\source\repos\ConsoleApp1\ConsoleApp1\ConsoleApp1.csproj

**Expected:** project builds successfully

**Actual:** Restore succeeds, but when it tries to build the project, it fails, looking for a project with some of the elements duplicated.  In my case:

> C:\Users\daplaist\source\repos\Users\daplaist\source\repos\ConsoleApp1\ConsoleApp1\ConsoleApp1.csproj : error MSB4025: The project file could not be loaded. Could not find a part of the path 'C:\Users\daplaist\source\repos\Users\daplaist\source\repos\ConsoleApp1\ConsoleApp1\ConsoleApp1.csproj'.

@rainersigwald @cdmihai @livarcocc

**Comment by rainersigwald** [comment]
That's worrisome, but I don't seem to be able to repro. Can you be even more specific in your repro steps? Anything special about that project, maybe?

**Comment by dsplaisted** [comment]
It's an Sdk-style project, if that helps.  I ran into this with a project in the core-sdk repo, so I tried reproing it on my ConsoleApp1, without looking to see what the contents of that project were. It looks like this:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <!--<TargetFramework>net471</TargetFramework>-->
    <TargetFramework>netcoreapp2.1</TargetFramework>
    <RuntimeIdentifiers>win7-x86;win7-x64</RuntimeIdentifiers>
  </PropertyGroup>

</Project>
```

**Comment by rainersigwald** [comment]
Yeah, still not seeing it even with that project file:

```
s:\msbuild2>s:\msbuild\artifacts\Debug\bootstrap\net472\MSBuild\Current\Bin\MSBuild.exe /restore ..\repro\Microsoft\msbuild\issues\3926\3926.csproj /v:m
Microsoft (R) Build Engine version 16.0.217-preview+gc71cbc0190 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 76.6 ms for s:\repro\Microsoft\msbuild\issues\3926\3926.csproj.
  3926 -> s:\repro\Microsoft\msbuild\issues\3926\bin\Debug\netcoreapp2.1\3926.dll
```

**Comment by rainersigwald** [comment]
I think/hope this was fixed by Microsoft/msbuild#4039. If you see it again, we should reopen and investigate more.
